File: Backend/mapping/src/main/java/com/staffinghub/coding/challenges/mapping/mappers/ArticleMapper.java
package com.staffinghub.coding.challenges.mapping.mappers;

import com.staffinghub.coding.challenges.mapping.models.db.Article;
import com.staffinghub.coding.challenges.mapping.models.dto.ArticleDto;
import org.springframework.stereotype.Component;

@Component
public class ArticleMapper {

    public ArticleDto map(Article article){
        //TODO
        return new ArticleDto();
    }

    public Article map(ArticleDto articleDto) {
        // Nicht Teil dieser Challenge.
        return new Article();
    }
}

File: Backend/mapping-kotlin/src/main/kotlin/com/staffinghub/coding/challenges/mapping/mappers/ArticleMapper.kt
package com.staffinghub.coding.challenges.mapping.mappers

import com.staffinghub.coding.challenges.mapping.models.db.Article
import com.staffinghub.coding.challenges.mapping.models.dto.ArticleDto
import org.springframework.stereotype.Component
import java.util.*

@Component
class ArticleMapper {
    fun map(article: Article?): ArticleDto {
        //TODO
        return ArticleDto(0, "", "", "", emptyList())
    }

    // Not part of the challenge / Nicht Teil dieser Challenge.
    fun map(articleDto: ArticleDto?): Article = Article(
        title = "An Article",
        blocks = emptySet(),
        id = 1,
        lastModified = Date()
    )
}
